### PR TITLE
Build with vcpkg manifest file and CMakePresets.txt

### PR DESCRIPTION
**Pull request by rsjtaylor**

### CODE REVIEW

**Comment by rsjtaylor** [comment]
The `workflow_dispatch` trigger is supposed to allow a manual CI build via the actions tab, but it's only enabled if set for a workflow in the default branch, so no way to check that it's working as expected until we merge.

**Comment by rsjtaylor** [comment]
Wondering if I should split the packaging preset into packaging (no ccache) and CI (with ccache)?
Might also be a good idea to only set the macos target os only in the packaging build, I'm not sure if that will fail if you don't have the 10.14 SDK installed.
Would adding a 'Developer' preset with Debug enabled and some additional debug flags also be useful?

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -134,38 +105,37 @@ jobs:
...
-          CCACHE_COMPRESSLEVEL: '6'
-          CCACHE_MAXSIZE: '500M'
-        uses: lukka/run-cmake@v3
-        with:
-          cmakeListsOrSettingsJson: CMakeListsTxtAdvanced
-          cmakeListsTxtPath: '${{ github.workspace }}/CMakeLists.txt'
-          useVcpkgToolchainFile: ${{ matrix.config.os != 'ubuntu*' }}
-          buildDirectory: '${{ runner.workspace }}/b/ninja'
-          cmakeAppendedArgs: '-DCMAKE_INSTALL_PREFIX="${{ steps.cmake_install_prefix.outputs.install_prefix }}" ${{ matrix.config.cmake_config }}'
+          VCPKG_BINARY_SOURCES: 'clear;files,${{ env.VCPKG_CACHE_LOCATION }},readwrite'
```
Comment (tomjnixon) [inline .github/workflows/build.yml:115]: the syntax of this is quite non-obvious; maybe worth a comment?

File: .gitignore [hunk 2]
```diff
@@ -1,4 +1,5 @@
 build*/
+cmake-build*/
```
Comment (tomjnixon) [inline .gitignore:2]: can't see this mentioned anywhere else?

Comment (rsjtaylor) [reply .gitignore:2]: Ah that snuck in by mistake, it's just where CLion puts build directories by default and not really anything to do with this PR.

File: .github/workflows/build.yml (hunk 1 above)
Comment (rsjtaylor) [reply .github/workflows/build.yml:115]: Good point, will add a link to the vcpkg docs that explain it as I missed it several times myself.

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -1,126 +1,97 @@
 name: Build
-on: [push, pull_request]
+on:
```
Comment (tomjnixon) [inline .github/workflows/build.yml:2]: Would it make sense to run this when pushing to a tag to create releases?

https://futurestud.io/tutorials/github-actions-run-a-workflow-when-creating-a-tag#runworkflowfornewtags

**Comment by tomjnixon** [comment]
> Wondering if I should split the packaging preset into packaging (no ccache) and CI (with ccache)?

maybe, it looks like the CI always does packaging though. Is there much advantage (speed / space) to not running packaging, except for tags, for example?

> Might also be a good idea to only set the macos target os in the packaging build, I'm not sure if that will fail if you don't have the 10.14 SDK installed.

It seems like a good idea to always build against the same SDK, so I'd leave it.

> Would adding a 'Developer' preset with Debug enabled and some additional warning flags also be useful?

Yes, perhaps in another PR unless you have it ready to go already?

Looks good in general to me though, and I now have a better feel for how presets work.

**Comment by rsjtaylor** [comment]
> > Wondering if I should split the packaging preset into packaging (no ccache) and CI (with ccache)?
> 
> maybe, it looks like the CI always does packaging though. Is there much advantage (speed / space) to not running packaging, except for tags, for example?

It was more that you might want to run packaging locally (Perhaps when sending some custom build off for someone to try out), and might not have ccache installed.

> 
> > Might also be a good idea to only set the macos target os in the packaging build, I'm not sure if that will fail if you don't have the 10.14 SDK installed.
> 
> It seems like a good idea to always build against the same SDK, so I'd leave it.

Yeah I that makes sense as you'd only really need to build from source if you want to work on the plugins, in which case we want consistency or at least a conscious choice to do something else.

> 
> > Would adding a 'Developer' preset with Debug enabled and some additional warning flags also be useful?
> 
> Yes, perhaps in another PR unless you have it ready to go already?

Good idea

> 
> Looks good in general to me though, and I now have a better feel for how presets work.

Thanks!

**Comment by tomjnixon** [comment]
> > > Wondering if I should split the packaging preset into packaging (no ccache) and CI (with ccache)?
> > 
> > 
> > maybe, it looks like the CI always does packaging though. Is there much advantage (speed / space) to not running packaging, except for tags, for example?
> 
> It was more that you might want to run packaging locally (Perhaps when sending some custom build off for someone to try out), and might not have ccache installed.

Ah right, yeah, that makes sense. Thinking about it again, I would be tempted to set up the ccache variables in the CI config, because it's really an environment-dependant thing (both the configuration and the dependency), and you've already got a bunch of code in the CI config to deal with it, so you might as well have it all in one place.

**Pull request merged**